src/components/fileviewer.js

```diff
@@ -15,88 +15,95 @@ export default class FileViewerContainer extends Component {
     };
-
-    this.setSelectedLine = this.setSelectedLine.bind(this)
-  }
-
-  componentDidMount() {
+    this.setSelectedLine = this.setSelectedLine.bind(this);
     /* get revision and path parameters from URL */
-    const parsedQuery = queryString.parse(this.props.location.search);
+    const parsedQuery = queryString.parse(props.location.search);
     if (!parsedQuery.revision || !parsedQuery.path) {
       this.setState({ appError: "Undefined URL query ('revision', 'path' fields are required)" });
     }
-    /* remove begining '/' in the path parameter */
+    /* remove beginning '/' in the path parameter */
     if (parsedQuery.path.startsWith('/')) {
       parsedQuery.path = parsedQuery.path.slice(1);
     }
-    this.revision = parsedQuery.revision
-    this.path = parsedQuery.path
+    this.revision = parsedQuery.revision;
+    this.path = parsedQuery.path;
+  }
 
+  componentDidMount() {
     /* Fetch source code from hg */
     FetchAPI.getRawFile(this.revision, this.path)
-      .then(text => this.setState({ parsedFile: text.split("\n") }))
+      .then(text => this.setState({ parsedFile: text.split('\n') }))
     ;
 
     /* Fetch coverages from ActiveData */
     FetchAPI.query({
-      "from": "coverage",
-      "where": {
-        "and": [
-          {"eq": {"source.file.name": `${this.path}`}},
-          {"eq": {"repo.changeset.id12": `${this.revision}`}}
-        ]
+      from: 'coverage',
+      where: {
+        and: [
+          { eq: { 'source.file.name': `${this.path}` } },
+          { eq: { 'repo.changeset.id12': `${this.revision}` } },
+        ],
       },
-      "limit": 1000,
-      "format": "list"
+      limit: 1000,
+      format: 'list',
     })
-    .then(data => {
-      console.log(data);
-      this.setState({coverage: data});
-      this.parseTestsCoverage(data.data);
-    });
+      .then((data) => {
+        this.parseCoverage(data.data);
+      });
   }
 
-  /* Parse data returns from ActiveData */
-  parseTestsCoverage(data) {
-    console.log(data.length);
-    var stat = [];
+  setSelectedLine(selectedLineNumber) {
+    this.setState({ selectedLine: selectedLineNumber });
+  }
 
-    // initial list for each coverable line
-    if (data.length > 0) {
-      const fileCov = data[0].source.file;
-      fileCov.covered.concat(fileCov.uncovered).forEach(line => {
-        stat[line] = [];
+  /* Parse coverage data */
+  parseCoverage(data) {
+    const covered = [];
+    const uncovered = [];
+    const testsPerHitLine = [];
+    const testsPerMissLine = [];
+
+    data.forEach((d) => {
+      d.source.file.covered.forEach((line) => {
+        covered.push(line);
+        if (!testsPerHitLine[line]) {
+          testsPerHitLine[line] = [];
+        }
+        testsPerHitLine[line].push(d);
       });
-    }
 
-    data.forEach(d => {
-      d.source.file.covered.forEach(line => {
-        stat[line].push(d);
+      d.source.file.uncovered.forEach((line) => {
+        if (!testsPerHitLine[line]) {
```

Comment:
I think adding this line may not fix the issue,
since after finish running the first ```forEach``` loop, if the first test does not cover a covered line, testsPerHitLine[line] will still be ```undefined```
If do a console log ```console.log(this.state.coverage.uncoveredLines)```, in the Platform.cpp example, the line 136, 138 are in the uncoveredLines list

I suggest we do not nest the second inner ```forEach``` loop in ```data.forEach()```
```
    data.forEach((d) => {
      d.source.file.covered.forEach((line) => {
        covered.push(line);
        if (!testsPerHitLine[line]) {
          testsPerHitLine[line] = [];
        }
        testsPerHitLine[line].push(d);
      });
    });

    data.forEach((d) => {
      d.source.file.uncovered.forEach((line) => {
        if (!testsPerHitLine[line]) {
          uncovered.push(line);
          if (!testsPerMissLine[line]) {
            testsPerMissLine[line] = [];
          }
          testsPerMissLine[line].push(d);
        }
      });
    });
```

Reply:
yes.. sorry I was doing a quick refactor and thought they were two separate loops.